framework/src/outputs/OutputWarehouse.C

```diff
@@ -314,11 +328,39 @@ OutputWarehouse::buildInterfaceHideVariables(const std::string & output_name,
 }
 
 void
-OutputWarehouse::checkOutputs(const std::set<OutputName> & names)
+OutputWarehouse::checkOutputs(const std::set<OutputName> & names,
+                              const bool supports_material_output)
 {
+  std::string reserved_name = "";
   for (const auto & name : names)
-    if (!isReservedName(name) && !hasOutput(name))
-      mooseError("The output object '", name, "' is not a defined output object");
+  {
+    const bool is_reserved_name = isReservedName(name);
+    if (is_reserved_name)
+      reserved_name = name;
+    if (!is_reserved_name)
+    {
+      if (!hasOutput(name))
+        mooseError("The output object '", name, "' is not a defined output object.");
+      if (supports_material_output && !hasMaterialPropertyOutput(name))
```

Comment:
supports_material_output is always true, I think you can get rid of it.

Reply:
Will be false in non-material calls in `CheckOutputAction`